src/main/java/com/truelayer/java/http/OkHttpClientFactory.java

```diff
@@ -40,7 +46,16 @@ public OkHttpClient buildBaseApiClient(
         OkHttpClient.Builder clientBuilder = new OkHttpClient.Builder();
 
         if (isNotEmpty(timeout)) {
+            // If the user sets a custom timeout, we set it to be the call timeout AND we set to 0 the default connect,
+            // read
+            // and write timeout values (0 means no timeout).
+            clientBuilder.connectTimeout(Duration.ZERO);
+            clientBuilder.readTimeout(Duration.ZERO);
+            clientBuilder.writeTimeout(Duration.ZERO);
```

Comment:
This was a 🐛 . A client setting a custom call timeout let's say at 30s, could have anyway hit the 10s imposed by one of connect, read, write timeouts coming with the OKHttp library defaults.

Reply:
Would it make sense to force clients to set a call timeout greater or equal than our default read timeout value? Or do we want to give the client complete freedom on it? In the latter case - if the custom client timeout is too short - we might end up with inconsistent situations like a payment being created on TL side resulting in a timeout error on client side. Not a big issue though given we should have idempotent endpoints so we should be able to handle retries, but I'm wondering if this default 30s value and consequences with custom lower values are clearly communicated to clients

Reply:
I thought about potential validations as well. My 2 cents here is that ultimately the only reason to use the `setTimeout(Duration timeout)` method on the builder is when a customer wants to specify an overall(call) timeout value which is lower that our maximum default (10s connect + 10s write + 30s read). They might want to do this to trade some consistency with reliability of their services I think. As such, I did not introduce any minimum value for a custom timeout value. And, as you said idempotency is there to help with those scenarios.

On the other side, It would be pretty much pointless to set a timeout value bigger than our max, as a request would be anyway cut by our ingress at the API level after ~30s. We could indeed introduce a validation to reject values bigger than 50s for the overall timeout I think, but I did not add that to not introduce too much coupling between our library and API (Perhaps we want to change this default in future on our API ingress). Moreover it would have been a _potential_ breaking change for existing customers. 

Do you agree ?

Reply:
Yes makes sense, I agree, thanks!